app/components-react/highlighter/StreamCard.tsx

```diff
@@ -0,0 +1,365 @@
+import React from 'react';
+import {
+  EAiDetectionState,
+  EHighlighterView,
+  IHighlightedStream,
+  IViewState,
+  StreamInfoForAiHighlighter,
+  TClip,
+} from 'services/highlighter';
+import styles from './StreamCard.m.less';
+import { Button } from 'antd';
+import { Services } from 'components-react/service-provider';
+import { isAiClip } from './utils';
+import { useVuex } from 'components-react/hooks';
+import { InputEmojiSection } from './InputEmojiSection';
+import { $t } from 'services/i18n';
+
+export default function StreamCard({
+  streamId,
+  clipsOfStreamAreLoading,
+  emitSetView,
+  emitGeneratePreview,
+  emitExportVideo,
+  emitRemoveStream,
+  emitCancelHighlightGeneration,
+}: {
+  streamId: string;
+  clipsOfStreamAreLoading: string | null;
+  emitSetView: (data: IViewState) => void;
+  emitGeneratePreview: () => void;
+  emitExportVideo: () => void;
+  emitRemoveStream: () => void;
+  emitCancelHighlightGeneration: () => void;
+}) {
+  const { HighlighterService } = Services;
+  const clips = useVuex(() =>
+    HighlighterService.views.clips
+      .filter(c => c.streamInfo?.[streamId])
+      .map(clip => {
+        if (isAiClip(clip) && (clip.aiInfo as any).moments) {
+          clip.aiInfo.inputs = (clip.aiInfo as any).moments;
+        }
+        return clip;
+      }),
+  );
+  const stream = useVuex(() =>
+    HighlighterService.views.highlightedStreams.find(s => s.id === streamId),
+  );
+  if (!stream) {
+    return <>error</>;
```

Comment:
```
  const stream = useVuex(() =>
    HighlighterService.views.highlightedStreams.find(s => s.id === streamId),
  );
  if (!stream) {
    return <>error</>;
  }
```

Its only there because the find() could technically return undefined - wich should never happen.
Are you thinking to make the error case "nicer" or check it differently?

Reply:
i see, in that case i would probably just recommend returning `<></>` since a blank screen is probably better than just some plaintext in terms of user visibility, the other option would be to pass through the whole stream as a prop instead of just the streamId